test/test_utils.go

```diff
@@ -281,3 +287,30 @@ func verifySignature(ctx context.Context, t *testing.T, c *clients, obj objects.
 		}
 	}
 }
+
+// restartChainsControllerPod restarts the pod running Chains
+// it then waits for a given timeout for the pod to resume running state
+func restartChainsControllerPod(c kubernetes.Interface, timeout time.Duration) error {
+	pods, err := c.CoreV1().Pods("tekton-chains").List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	pod := pods.Items[0]
+	podUid := pod.UID
+	err = c.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	return wait.PollUntilContextTimeout(context.Background(), 2*time.Second, timeout, true, func(context.Context) (done bool, err error) {
+		pods, err := c.CoreV1().Pods("tekton-chains").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		pod := pods.Items[0]
+		if pod.UID != podUid {
```

Comment:
I don't think this will ever return true - a newly created pod should always have a new UID.

In fact, you may want to check the opposite - if it is the same UID then it means that we're still on the existing pod and we're not done waiting yet.

Reply:
Fixed